java/integTest/src/test/java/glide/SharedCommandTests.java

```diff
@@ -4436,4 +4436,45 @@ public void blmove_timeout_check(BaseClient client) {
                                     .get(3, TimeUnit.SECONDS));
         }
     }
+
+    @SneakyThrows
+    @ParameterizedTest(autoCloseArguments = false)
+    @MethodSource("getClients")
+    public void sintercard(BaseClient client) {
+        assumeTrue(REDIS_VERSION.isGreaterThanOrEqualTo("7.0.0"), "This feature added in redis 7.0.0");
+        // setup
+        String key1 = "{key}-1" + UUID.randomUUID();
+        String key2 = "{key}-2" + UUID.randomUUID();
+        String nonSetKey = "{key}-4" + UUID.randomUUID();
+        String[] saddargs = {"one", "two", "three", "four"};
+        String[] saddargs2 = {"two", "three", "four", "five"};
+        long limit = 2;
+        long limit2 = 4;
+
+        // keys does not exist or is empty
+        String[] keys = new String[] {key1, key2};
+        assertEquals(0, client.sintercard(keys).get());
+        assertEquals(0, client.sintercard(keys, limit).get());
+
+        // one of the keys is empty, intersection is empty, cardinality equals to 0
+        assertEquals(4, client.sadd(key1, saddargs).get());
+        assertEquals(0, client.sintercard(keys).get());
+
+        // sets at both keys have value, get cardinality of the intersection
+        assertEquals(4, client.sadd(key2, saddargs2).get());
+        assertEquals(3, client.sintercard(keys).get());
+
+        // returns limit as cardinality when the limit is reached partway through the computation
+        assertEquals(limit, client.sintercard(keys, limit).get());
+
+        // returns cardinality normally when the limit isn't reached
+        assertEquals(3, client.sintercard(keys, limit2).get());
```

Comment:
you have this test above with the empty list and limit (line 4457).  Feel free to remove this test.